woocommerce/woocommerce-android
# Issue/3725 check bluetooth

Parent issue #3725 

Merge instructions:
1. Make sure https://github.com/woocommerce/woocommerce-android/pull/4007 is merged
2. Remove "do not merge" label
3. Merge this PR

This PR adds bluetooth-enabled check when the user enters "connect reader" flow. 

Logic summary
- bluetooth is enabled, start the scan
- bluetooth is disabled, request "enable-bluetooth"
- user allows the request (aka enables bluetooth), start the scan
- user denies the request (aka doesn't enable bluetooth), show "BluetoothDisabled" error screen

To test
- disable bluetooth (bluetooth is not available on emulators so you'll need to use a phone)
1. Open app settings
2. Tap on "Manage card reader
3. Tap on "Redirect to CardReaderDetailFragment"
4. Tap on "Connect"
5. Notice "enable bluetooth system dialog" is shown
6. Deny the request
7. Notice "Bluetooth disable" error shown
8. Tap on "Open App settings"
9. Notice "enable bluetooth system dialog" is shown
10. Allow the request and notice the connection flow starts

- Repeat steps 1 - 5 but give the apps the permissions in step 6



Update release notes:
